scanspec/regions.py

```diff
@@ -242,6 +242,55 @@ def mask(self, points: AxesPoints) -> np.ndarray:
         return mask_x & mask_y
 
 
+@dataclass
+class Polygon(Region):
+    """Mask contains points of axis within a rotated xy polygon
+    .. example_spec::
+
+        from scanspec.specs import Line
+        from scanspec.regions import Polygon
+        grid = Line("y", 1, 3, 10) * ~Line("x", 0, 2, 10)
+        spec = grid & Polygon("x", "y", x_verts_array, y_verts_array)
+    """
+
+    x_axis: str = field(
+        metadata=schema(description="The name matching the x axis of the spec")
+    )
+    y_axis: str = field(
+        metadata=schema(description="The name matching the y axis of the spec")
+    )
+    x_verts: np.ndarray = field(
+        metadata=schema(
+            description="The Nx1 x coordinates of the polygons vertices", min_len=4
+        )
+    )
+    y_verts: np.ndarray = field(
+        metadata=schema(
+            description="The Nx1 y coordinates of the polygons vertices", min_len=4
+        )
+    )
```

Comment:
Yes, I believe these get converted to ndarrays inside of matplotlib's patch functions. Have since changed this and they can be plotted without any problems